app/driver_flexvolume.go

```diff
@@ -0,0 +1,120 @@
+package app
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/rancher/longhorn-manager/types"
+)
+
+const (
+	DetectFlexVolumeName   = "discover-flexvolume-dir"
+	DetectFlexVolumeScript = `
+    find_kubelet_proc() {
+      for proc in $(find /proc -maxdepth 1 -type d 2>/dev/null); do
+        if [ ! -f $proc/cmdline ]; then
+          continue
+        fi
+        if [[ "$(cat $proc/cmdline | tr '\000' '\n' | head -n1 | tr '/' '\n' | tail -n1)" == "kubelet" ]]; then
+          echo $proc
+          return
+        fi
+      done
+    }
+    get_flexvolume_path() {
+      proc=$(find_kubelet_proc)
+      if [ "$proc" != "" ]; then
+        path=$(cat $proc/cmdline | tr '\000' '\n' | grep volume-plugin-dir | tr '=' '\n' | tail -n1)
+        if [ "$path" == "" ]; then
+          echo '/usr/libexec/kubernetes/kubelet-plugins/volume/exec/'
+        else
+          echo $path
+        fi
+        return
+      fi
+      # no kubelet process found, echo nothing
+    }
+    get_flexvolume_path
+  `
+)
+
+func discoverFlexvolumeDir(kubeClient *clientset.Clientset, managerImage string) (string, error) {
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		return "", fmt.Errorf("Cannot detect pod namespace, environment variable %v is missing", types.EnvPodNamespace)
+	}
+
+	if err := deployPod(kubeClient, namespace, managerImage); err != nil {
+		logrus.Warnf("Failed to deploy flexvolume detection pod: %v", err)
+		return "", err
+	}
+
+	defer func() {
+		if err := kubeClient.CoreV1().Pods(namespace).Delete(DetectFlexVolumeName, &metav1.DeleteOptions{}); err != nil {
+			logrus.Warnf("Failed to delete flexvolume detection pod: %v", err)
+		}
+	}()
+
+	for delay := 1 * time.Second; ; delay *= 2 {
```

Comment:
Don't wait indefinitely.

Also, the exponentially back off is too long in the second level. Just do a normal 1-second check for a certain period of time is enough.